wtforms/fields/core.py

```diff
@@ -940,11 +940,16 @@ def validate(self, form, extra_validators=tuple()):
         for subfield in self.entries:
             if not subfield.validate(form):
                 self.errors.append(subfield.errors)
+            else:
```

Comment:
Collect the subfield errors unconditionally to ensure that they're all references to the error lists, even when empty.

```python
subfield.validate(form)
self.errors.append(subfield.errors)
```